File: status-resolver/src/main/java/com/ackee/versioupdatehandler/VersionFetcher.java
package com.ackee.versioupdatehandler;

import com.ackee.versioupdatehandler.model.VersionsConfiguration;

import io.reactivex.Single;


/**
 * Base interface for classes that will handle version fetching.
 */
public interface VersionFetcher {
    public static final String TAG = VersionFetcher.class.getName();

    Single<VersionsConfiguration> fetch();

}

File: updatechecker-status-resolver/src/main/java/io/github/ackeecz/versionupdatehandler/VersionFetcher.kt
package io.github.ackeecz.versionupdatehandler

import io.github.ackeecz.versionupdatehandler.model.VersionsConfiguration

/**
 * Base interface for classes that will handle version fetching.
 */
interface VersionFetcher {

    suspend fun fetch(): VersionsConfiguration
}
